solr/core/src/test/org/apache/solr/search/join/BlockJoinChildQueryParserTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.search.join;
+
+import org.apache.solr.SolrTestCaseJ4;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class BlockJoinChildQueryParserTest extends SolrTestCaseJ4 {
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initCore("solrconfig.xml", "schema.xml");
+        createIndex();
+    }
+
+    public static void createIndex() throws IOException, Exception {
+        assertU(adoc("id", "1", "name", "45", "directed_by_s", "Lennon"));
+        assertU(adoc("id", "2", "name", "42", "directed_by_s", "Block"));
+        assertU(commit());
+        assertQ(req("q", "*:*"), "//*[@numFound='" + 2 + "']");
+    }
+
+    @Test
+    public void testChildrenParserEmpty() {
+        assertQ(
+                req("q", "{!child q={}"), "//*[@numFound='2']");
```

Comment:
Could this be in `BJQParserTest` instead of creating a new class for a single assertion?